app/services/user_service/user_signin.rb

```diff
@@ -7,19 +7,10 @@ def initialize(args)
 
         def call
             user = User.find_by(email: @email)
-            if user.nil?
-                return {
-                success: false,
-                message: "User not found",
-                }
-            end
+            raise StandardError, "User Not Exists" if user.nil?
```

Comment:
그냥 StandardError를 raise해주는 것보다,
서비스 최상단에 이런 식으로 작성해보는 것이 어떨까요
```
module UserService
    class UserSignin < ApplicationService
       class ValidationError < StandardError; end
```
이렇게 하고 나서 여기서는 
```suggestion
            raise ValidationError, "User Not Exists" if user.nil?
```
를 raise해주면, 이를 핸들하는 곳에서
```
rescue UserService::UserSignin::ValidationError => e
```
로 처리할 수 있습니다. 익셉션 처리를 할 때엔 스코프를 좁힐  수 있으면 좋으니깐요

Reply:
> 그냥 StandardError를 raise해주는 것보다, 서비스 최상단에 이런 식으로 작성해보는 것이 어떨까요
> 
> ```
> module UserService
>     class UserSignin < ApplicationService
>        class ValidationError < StandardError; end
> ```
> 
> 이렇게 하고 나서 여기서는
> 
> 를 raise해주면, 이를 핸들하는 곳에서
> 
> ```
> rescue UserService::UserSignin::ValidationError => e
> ```
> 
> 로 처리할 수 있습니다. 익셉션 처리를 할 때엔 스코프를 좁힐 수 있으면 좋으니깐요

요러한 방식으로 수정해보시겠어요?

Reply:
넵!!